generator/testdata/dart/protobuf/golden/secretmanager/lib/secretmanager.dart

```diff
@@ -0,0 +1,1005 @@
+  final String? name;
+
+  Topic({
+    this.name,
+  });
+}
+
+/// The rotation time and period for a
+/// [Secret][google.cloud.secretmanager.v1.Secret]. At next_rotation_time, Secret
+/// Manager will send a Pub/Sub notification to the topics configured on the
+/// Secret. [Secret.topics][google.cloud.secretmanager.v1.Secret.topics] must be
+/// set to configure rotation.
+class Rotation {
+
+  /// Optional. Timestamp in UTC at which the
+  /// [Secret][google.cloud.secretmanager.v1.Secret] is scheduled to rotate.
+  /// Cannot be set to less than 300s (5 min) in the future and at most
+  /// 3153600000s (100 years).
+  /// 
+  /// [next_rotation_time][google.cloud.secretmanager.v1.Rotation.next_rotation_time]
+  /// MUST  be set if
+  /// [rotation_period][google.cloud.secretmanager.v1.Rotation.rotation_period]
+  /// is set.
+  final DateTime? nextRotationTime;
+
+  /// Input only. The Duration between rotation notifications. Must be in seconds
+  /// and at least 3600s (1h) and at most 3153600000s (100 years).
+  /// 
+  /// If
+  /// [rotation_period][google.cloud.secretmanager.v1.Rotation.rotation_period]
+  /// is set,
+  /// [next_rotation_time][google.cloud.secretmanager.v1.Rotation.next_rotation_time]
+  /// must be set.
+  /// [next_rotation_time][google.cloud.secretmanager.v1.Rotation.next_rotation_time]
+  /// will be advanced by this period when the service automatically sends
+  /// rotation notifications.
+  final Duration? rotationPeriod;
+
+  Rotation({
+    this.nextRotationTime,
+    this.rotationPeriod,
+  });
+}
+
+/// A secret payload resource in the Secret Manager API. This contains the
+/// sensitive secret payload that is associated with a
+/// [SecretVersion][google.cloud.secretmanager.v1.SecretVersion].
+class SecretPayload {
+
+  /// The secret data. Must be no larger than 64KiB.
+  final Uint8List? data;
+
+  /// Optional. If specified,
+  /// [SecretManagerService][google.cloud.secretmanager.v1.SecretManagerService]
+  /// will verify the integrity of the received
+  /// [data][google.cloud.secretmanager.v1.SecretPayload.data] on
+  /// [SecretManagerService.AddSecretVersion][google.cloud.secretmanager.v1.SecretManagerService.AddSecretVersion]
+  /// calls using the crc32c checksum and store it to include in future
+  /// [SecretManagerService.AccessSecretVersion][google.cloud.secretmanager.v1.SecretManagerService.AccessSecretVersion]
+  /// responses. If a checksum is not provided in the
+  /// [SecretManagerService.AddSecretVersion][google.cloud.secretmanager.v1.SecretManagerService.AddSecretVersion]
+  /// request, the
+  /// [SecretManagerService][google.cloud.secretmanager.v1.SecretManagerService]
+  /// will generate and store one for you.
+  /// 
+  /// The CRC32C value is encoded as a Int64 for compatibility, and can be
+  /// safely downconverted to uint32 in languages that support this type.
+  /// https://cloud.google.com/apis/design/design_patterns#integer_types
+  final int? dataCrc32C;
```

Comment:
BTW, if you ever need to snake-case this: beware because the original name is `data_crc32c` but the snake_case -> camelCase -> snake_case conversion breaks. No action needed.